app/scripts/modules/core/src/presentation/hooks/useData.hook.ts

```diff
@@ -1,23 +1,26 @@
 import { IPromise } from 'angular';
-import { $q } from 'ngimport';
 import { isNil } from 'lodash';
 
 import { useLatestPromise, IUseLatestPromiseResult } from './useLatestPromise.hook';
 
 /**
- * A react hook which invokes a promise returning callback whenever any of its dependencies changes.
- * Returns the provided default value whenever any dependency is null or undefined
+ * A react hook which invokes a promise factory callback whenever any of its dependencies changes.
+ *
+ * Returns the default result until one promise resolves.
```

Comment:
To be consistent with L17:
```suggestion
 * Returns the default result until the first promise resolves.
```